### DISCUSSION TITLE
physical button for quicksnap

### DISCUSSION THREAD

**Discussion by vogues** [JulienHeijmans/quicksnap#13] [Ideas]
Would you please add a quicksnap button?

I'm using serpens and am trying to create a custom pie menu with your addon.

**Comment by JulienHeijmans** [comment]
Could you explain a bit more, how does having a button help with serpens addon?

The tool is based on a modal operator:  `object.quick_vertex_snap` (I will probably rename it to `object.quicksnap`  in a next update, it is still the old name of the tool), I am not sure how adding a button to this operator will help.

If you are looking for a "WorkSpaceTool", like all the buttons on the left column, I looked into it and it sounded complicated, as they are not really linked to a modal operator, it looks like they more made to display gizmos/widgets that themselves trigger operators, I can't do that with this tool. But I can look more into other ways to do that...  (I am still not sure if this is what you are looking for.)

**Reply by vogues** [reply]
> Could you explain a bit more, how does having a button help with serpens addon?
> 
> The tool is based on a modal operator: `object.quick_vertex_snap` (I will probably rename it to `object.quicksnap` in a next update, it is still the old name of the tool), I am not sure how adding a button to this operator will help.
> 
> If you are looking for a "WorkSpaceTool", like all the buttons on the left column, I looked into it and it sounded complicated, as they are not really linked to a modal operator, it looks like they more made to display gizmos/widgets that themselves trigger operators, I can't do that with this tool. But I can look more into other ways to do that... (I am still not sure if this is what you are looking for.)

I was looking for an N panel button On the right hand column.

**Reply by JulienHeijmans** [reply]
Oh, yes, a N-Panel is definitely doable. (Probably disabled by default, and an option would enable it in the settings)
Any suggestion where you would see that button be? A new n panel with a single button, or maybe in the existing Tool panel?

**Reply by vogues** [reply]
> Oh, yes, a N-Panel is definitely doable. (Probably disabled by default, and an option would enable it in the settings)
> Any suggestion where you would see that button be? A new n panel with a single button, or maybe in the existing Tool panel?

Personally, ill be integrating it into a pie via serpens, so location doesn't matter to me.

It may be nice to have on the top bar for other users, or in a frequently used panel.

**Reply by JulienHeijmans** [reply]
Could you explain to me again why you need a button to be able to use quicksnap in serpens?
Can you not just invoke the `object.quick_vertex_snap`  operator?
How does a button help?

**Reply by vogues** [reply]
> Could you explain to me again why you need a button to be able to use quicksnap in serpens? Can you not just invoke the `object.quick_vertex_snap` operator? How does a button help?

possibly- im very new to coding/visual coding. this request is mostly out of my lack of knowledge.
I'll try that when I have a chance. 

Thank you!

**Reply by vogues** [reply]
> Could you explain to me again why you need a button to be able to use quicksnap in serpens? Can you not just invoke the `object.quick_vertex_snap` operator? How does a button help?

I've tried both to no avail:
bpy.ops.object.quicksnap()
bpy.ops.object.quick_vertex_snap

I'm pretty sure serpens hooks into the panel since commands like update from the options menu link properly.

**Comment by vogues** [comment]
Thanks for the follow up btw- this is a very cool feature you've developed
